tests/integration_tests/test_ds_identify.py

```diff
@@ -0,0 +1,38 @@
+"""test that ds-identify works as expected"""
+
+import json
+
+from tests.integration_tests.instances import IntegrationInstance
+from tests.integration_tests.integration_settings import OS_IMAGE, PLATFORM
+from tests.integration_tests.util import verify_clean_log, wait_for_cloud_init
+
+DATASOURCE_LIST_FILE = "/etc/cloud/cloud.cfg.d/90_dpkg.cfg"
+MAP_PLATFORM_TO_DATASOURCE = {
+    "lxd_container": "lxd",
+    "lxd_vm": "lxd",
+    "qemu": "nocloud",
+    "ec2": "aws",
+    "oci": "oracle",
```

Comment:
I tested that this works for oracle, qemu, as well as ibm
```suggestion
    "oci": "oracle",
    "ibm": "nocloud",
```

Reply:
@a-dubs nice thanks!

I tested on ec2, gce, lxd, and have some educated guesses based on the code that it should work on azure and openstack (but can't test openstack currently).